drake/systems/System.h

```diff
@@ -0,0 +1,242 @@
+    ///           the x argument must be left out if the system has zero state (RowsAtCompileTime==0)
+    ///           the u argument must be left out if the system has no inputs or if the system is labeled as not direct feedthrough
+    /// Note that (by convention) the output method is called even if there is no output for the system. (e.g. for Visualizers)
+
+//  todo: implemented the templated forms here. (need to help compiler figure out the instantiation for casting to Eigen::Matrix<double>)
+    template <typename ScalarType>
+    OutputVector<ScalarType> output(const ScalarType& t, const StateVector<ScalarType>& x, const InputVector<ScalarType>& u) const {
+      return OutputDispatch< isTimeVarying, (num_states != 0), ((num_inputs != 0) && isDirectFeedthrough), ScalarType, Derived, StateVector, InputVector, OutputVector>::eval(static_cast<const Derived*>(this),t,x,u);
+    };
+/*
+    OutputVector<double> output(const StateVector<double>& x, const InputVector<double>& u) const {
+      static_assert(!isTimeVarying,"You must set the isTimeVarying template parameter to false to use this method");
+      return static_cast<const Derived*>(this)->outputImplementation(x,u);
+    };
+    OutputVector<double> output(const double& t, const StateVector<double>& x) const {
+      static_assert(!isDirectFeedthrough,"You must set the isDirectFeedthrough template parameter to false to use this method");
+      return static_cast<const Derived*>(this)->outputImplementation(t,x);
+    };
+    OutputVector<double> output(const StateVector<double>& x) const {
+      static_assert(!isTimeVarying,"You must set the isTimeVarying template parameter to false to use this method");
+      static_assert(!isDirectFeedthrough,"You must set the isDirectFeedthrough template parameter to false to use this method");
+      return static_cast<const Derived*>(this)->outputImplementation(x);
+    };
+*/
+
+    // todo: add sparsity information about dynamics, update, and output methods
+
+  };
+
+// This is the class design that I want.  But it seems that recursive templates in the inheritance line don't work (more discussion in the git history)
+//
+//  template <class System1, class System2>
+//  class FeedbackSystem : public System<FeedbackSystem<System1,System2>,
+//          CombinedVectorBuilder<System1::StateVectorType , System2::StateVectorType>::VecType, System1::InputVectorType, System1::OutputVectorType, System1::isTimeVarying||System2::isTimeVarying, System1::isDirectFeedthrough> {
+//
+// So I'm getting around it with this helper macro (which makes the call outside of the template class, so it's not recursive)
+#define FeedbackSystemType(System1,System2) Drake::FeedbackSystem<System1,System2,System1::StateVectorType,System2::StateVectorType,Drake::CombinedVectorBuilder<System1::StateVectorType,System2::StateVectorType>::VecType,System1::InputVectorType,System1::OutputVectorType,System1::isTimeVarying||System2::isTimeVarying,System1::isDirectFeedthrough>
```

Comment:
I'm going to take a crack at doing this without a macro. A major issue with this macro is currently that if System1 has multiple type parameters, it will look like `Foo<Bar, Baz>`. The comma will be interpreted in the macro expansion as the argument separator, so it will think that `, Baz>` is System2 and that you're passing in more arguments than it expects.